core/src/main/java/hudson/model/ViewDescriptor.java

```diff
@@ -101,13 +110,65 @@ public AutoCompletionCandidates doAutoCompleteCopyNewItemFrom(@QueryParameter fi
      * Possible {@link ListViewColumnDescriptor}s that can be used with this view.
      */
     public List<Descriptor<ListViewColumn>> getColumnsDescriptors() {
-        return ListViewColumn.all();
+        StaplerRequest request = Stapler.getCurrentRequest();
+        View view = request.findAncestorObject(clazz);
+        return view == null ? DescriptorVisibilityFilter.applyType(clazz, ListViewColumn.all())
+                : DescriptorVisibilityFilter.apply(view, ListViewColumn.all());
     }
 
     /**
      * Possible {@link ViewJobFilter} types that can be used with this view.
      */
     public List<Descriptor<ViewJobFilter>> getJobFiltersDescriptors() {
-        return ViewJobFilter.all();
+        StaplerRequest request = Stapler.getCurrentRequest();
+        View view = request.findAncestorObject(clazz);
+        return view == null ? DescriptorVisibilityFilter.applyType(clazz, ViewJobFilter.all())
+                : DescriptorVisibilityFilter.apply(view, ViewJobFilter.all());
     }
+
+    /**
+     * Validation of the display name field.
+     *
+     * @param view the view to check the new display name of.
+     * @param value the proposed new display name.
+     * @return the validation result.
+     * @since FIXME
+     */
+    @SuppressWarnings("unused") // expose utility check method to subclasses
```

Comment:
Yes but it needs to use the `Referer` header hack to resolve the `View` so probably waiting on dedicated support in stapler for that